### ISSUE TITLE
Issue with binary sampler

### ISSUE THREAD

**Issue by vandenman** [nimble-dev/nimble#386]
Hi, I encountered an issue with the binary sampler. That is, it gives the wrong result. The slice sampler does give the correct result. Thoughts?

```
library(nimble)

# the model: is the mean b[1] or b[2]?
modelBern <- nimbleCode(code = {
	
	indicator ~ dbern(0.5)
	mu <- indicator * b[1] + (1 - indicator) * b[2]
	for (i in 1:n) {
		
		x[i] ~ dnorm(mu, 1)
		
	}
	
})

# simulate data
set.seed(31415)
b = c(1e3, 5)
n = 1e3
x = rnorm(n, b[2], 1) # mean is b[2], indicator should be 0.
mean(x) # 5.01399

rModel <- nimbleModel(code = modelBern,
				  data = list(x = x),
				  constants = list(b = b, n = n),
				  dimensions = list(b = 2),
				  calculate = TRUE)
MCMCconf <- configureMCMC(model = rModel)
MCMCconf$printSamplers()

bMCMC <- buildMCMC(conf = MCMCconf)
cModel <- compileNimble(rModel)
mcmcObjBern <- compileNimble(bMCMC, project = cModel, resetFunctions = TRUE)

samplesBern <- runMCMC(mcmc = mcmcObjBern, niter = 1e3, nchains = 2, nburnin = 1e2, returnCodaMCMC = TRUE)
plot(samplesBern) # should be 0, but it is 1
colMeans(as.matrix(samplesBern)) # should be 0, but it is 1

# now with slice sampler

MCMCconf <- configureMCMC(model = rModel)

MCMCconf$removeSamplers("indicator")
MCMCconf$addSampler(target = 'indicator', type = 'slice')
MCMCconf$printSamplers()

bMCMC <- buildMCMC(conf = MCMCconf)
mcmcObjBern <- compileNimble(bMCMC, project = cModel, resetFunctions = TRUE)

samplesBern <- runMCMC(mcmc = mcmcObjBern, niter = 1e3, nchains = 2, nburnin = 1e2, returnCodaMCMC = TRUE)
plot(samplesBern) # should be 0 and is 0
colMeans(as.matrix(samplesBern)) # should be 0 and is 0
```

**Comment by danielturek** [comment]
@vandenman thanks for providing a reproducible example.  We'll look into this soon, and get back to you.

**Comment by perrydv** [comment]
This issue results from numerical underflow.  A proposed fix is in pull request #387.  For immediate purposes @vandenman you can install from branch fixIssue386_binary_sampler

**Comment by vandenman** [comment]
Thanks for the quick fix! I did some tests and the branch fixIssue386_binary_sampler seems to solve my problem.

**Comment by perrydv** [comment]
This is now on branch `devel` and will be released in version 0.6-5 soon (next week)
